src/App.tsx

```diff
@@ -1,24 +1,284 @@
+          )}
+
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={todoTitle}
+              onChange={(e) => setTodoTitle(e.target.value)}
+              disabled={tempTodo !== null}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main">
+          {visibleTodos?.map((todo) => (
+            <div
+              key={todo.id}
+              className={cn('todo', {
+                completed: todo?.completed,
+              })}
+            >
+              <label className="todo__status-label">
+                <input
+                  type="checkbox"
+                  className="todo__status"
+                  defaultChecked={todo?.completed}
+                />
+              </label>
+
+              <span className="todo__title">{todo?.title}</span>
+
+              <button
+                type="button"
+                className="todo__remove"
+                onClick={() => handleRemoveTodo(todo.id)}
+              >
+                ×
+              </button>
+
+              <div
+                className={cn('modal', 'overlay', {
+                  'is-active': isLoading && todo.id === 0,
+                })}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          ))}
+
+          {tempTodo && (
+            <div
+              key={tempTodo.id}
+              className={cn('todo', {
+                completed: tempTodo?.completed,
+              })}
+            >
+              <label className="todo__status-label">
+                <input
+                  type="checkbox"
+                  className="todo__status"
+                  defaultChecked={tempTodo?.completed}
+                />
+              </label>
+
+              <span className="todo__title">{tempTodo?.title}</span>
+
+              <button type="button" className="todo__remove">
+                ×
+              </button>
+
+              <div
+                className={cn('modal', 'overlay', {
+                  'is-active': isLoading,
+                })}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          )}
+        </section>
+
+        {todos?.length > 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">{`${uncompletedTodos?.length} items left`}</span>
+
+            <nav className="filter">
+              <a
+                href="#/"
+                className={cn('filter__link', {
+                  selected: filterStatus === FilterStatus.ALL,
+                })}
+                onClick={() => setFilterStatus(FilterStatus.ALL)}
```

Comment:
```suggestion
                onClick={makeSetFilterStatus(FilterStatus.ALL)}
```
instead to make arrow function in jsx better to make one function factory which will take filter and return arrow function and use it  for all cases
```
  const makeSetFilterStatus = (filter) => () => setFilterStatus(filter)
```